src/Mod/Part/BasicShapes/SkipGui.py

```diff
@@ -0,0 +1,267 @@
+#! python
+# -*- coding: utf-8 -*-
+# SkipGui.py
+# 2022, by Mark Ganson <TheMarkster>
+# LGPL 2.1 or later
+
+# this file is called from c++ in Mod/Part/Gui/Command.cpp
+# from various functions when these commands are activated:
+#   Part::Loft
+#   Part::Sweep
+#   Part::Extrude
+#   Part::Revolve
+# Then if the user has preselected certain objects and subobjects
+# the Gui can be skipped and the document objects directly created
+
+# If False is returned, then the Gui dialog should be shown
+# Else if True is returned, then the dialog can be skipped
+
+import FreeCAD as App
+import FreeCADGui as Gui
+
+def getActStr(document):
+    return f"App.getDocument('{document.Name}').ActiveObject"
+
+def getDocStr(document):
+    return f"App.getDocument('{document.Name}')"
+
+def getObjStr(document, object):
+    return f"{getDocStr(document)}.getObject('{object.Name}')"
+
+def getVisStr(document, objs):
+    ret = ""
+    for obj in objs:
+        ret += f"{getDocStr(document)}.getObject('{obj.Name}').ViewObject.Visibility = False\n"
+    return ret
+
+def getSetPropListStr(document, propName, objList):
+    ret = f"{getActStr(document)}.{propName} = ["
+    for obj in objList:
+        ret += f"{getDocStr(document)}.getObject('{obj.Name}'), "
+    ret += "]\n"
+    return ret
+
+def getAddObjStr(document, typeId, name):
+    ret = getDocStr(document)
+    ret += f".addObject('{typeId}', '{name}')\n"
+    return ret
+
+
+def makeExtrude():
+
+    selx = Gui.Selection.getSelectionEx()
+    if len(selx) == 0:
+        App.Console.PrintLog(f"makeExtrude() nothing selected, not skipping dialog\n")
+        return False
+
+    selected = [sel.Object for sel in selx if bool(sel.Object.isDerivedFrom("Part::2DObject") or
+                sel.Object.isDerivedFrom("Part::Face") or
+                sel.Object.isDerivedFrom("Part::FeaturePython") or
+                bool(hasattr(sel.Object,"Shape") and len(sel.Object.Shape.Vertexes) == 1) or
+                sel.Object.isDerivedFrom("Sketcher::SketchObject")) and
+                sel.HasSubObjects == False]
```

Comment:
Should that be took into account ? It will force user to take care of selecting the whole object rather than just one of it's subelements ? I have no strong opinion on this.